crates/turborepo-lib/src/config/user.rs

```diff
@@ -157,6 +158,18 @@ mod test {
             .with_environment(Some(env))
             .load()?;
         assert_eq!(config.token(), Some("bar"));
+
+        let mut config_file = NamedTempFile::new()?;
+        writeln!(&mut config_file, "{{\"token\": \"baz\"}}")?;
+        let env = {
+            let mut map = HashMap::new();
+            map.insert("VERCEL_ARTIFACTS_TOKEN".into(), "bat".into());
+            map
+        };
+        let config = UserConfigLoader::new(config_file.path().to_path_buf())
+            .with_environment(Some(env))
+            .load()?;
+        assert_eq!(config.token(), Some("bat"));
```

Comment:
Nit, but this seems like we should either refactor this into a function to be shared across tests or make this a table test instead of copy-pasta for each new environment override